app/src/lib/api.ts

```diff
@@ -292,10 +292,10 @@ export class API {
         private: private_,
       })
 
-      return parsedResponse<IAPIRepository>(response)
+      return await parsedResponse<IAPIRepository>(response)
     } catch (e) {
-      log.warn(`createRepository: failed with endpoint ${this.endpoint}`, e)
-      throw e
+      log.error(`createRepository: failed with endpoint ${this.endpoint}`, e)
+      throw new Error(`Unable to publish repository. Please check you have an internet connection and try again.`)
```

Comment:
But I think we want to rethrow the original error, because at this point it's `APIError` which has the server error JSON attached to it. So if we don't rethrow it we're throwing (ha) away context.